### PR TITLE
Adding fairea tool

### PR DESCRIPTION
**Pull request by fracarfer5**
Adding fairness-accuracy trade-off analyser: Fairea. Adding object and tutorial

### CODE REVIEW

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        """
+        return self.baseline_acc, self.baseline_fairness
+
+    def __mutate_preds(self, preds, to_mutate, ids, o):
+        """
+        Mutates the predictions of the model.
+
+        Parameters
+        ----------
+        preds: array, shape=(n_samples)
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:128]: (n_samples,)

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+
+    def __create_baseline(self, x, y, group_a, group_b):
+        """
+        Creates the baseline model for the dataset by applying the mutations.
+
+        Parameters
+        ----------
+        x: array, shape=(n_samples, n_features)
+            input data
+        y: array, shape=(n_samples)
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:157]: (n_samples,)

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+        (
+            X_train,
+            X_test,
+            y_train,
+            y_test,
+            _,
+            group_a_ts,
+            _,
+            group_b_ts,
+        ) = train_test_split(x, y, group_a, group_b, test_size=0.3, random_state=42)
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:178]: test_size as input parameter

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+
+        ids = [x for x in range(len(y_test))]
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:180]: range(int(0.3*len(y_test))) verify!

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+                    accuracy metric to be used: accuracy (acc) or roc-auc (auc)
+        data_splits: int
+                    number of splits to be used for cross-validation
+        repetitions: int
+                    number of repetitions of mutation to be performed
+        odds: dict
+                dictionary of odds to be used for mutation
+        options: list
+                list of options to be used for mutation
+        degrees: int
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:63]: improve documetnation

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+
+        acc_base = np.array(
+            [np.mean([row[0] for row in results["0"][degree]]) for degree in degrees]
+        )
+        fair_base = np.array(
+            [np.mean([row[1] for row in results["0"][degree]]) for degree in degrees]
+        )
+        return acc_base, fair_base
+
+    def add_scores(self, model_name, y_true, y_pred, group_a_ts, group_b_ts):
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:240]: group_a groupb

**Review by crismunoz — CHANGES_REQUESTED** [review]
Remove Docs
